src/libs/ReportUtils.js

```diff
@@ -851,6 +859,108 @@ function isUnread(report) {
     return lastReadSequenceNumber < maxSequenceNumber;
 }
 
+/**
+ * Determines if a report has an outstanding IOU that doesn't belong to the currently logged in user
+ *
+ * @param {Object} report
+ * @param {String} report.iouReportID
+ * @param {boolean} report.    if (!report || !report.iouReportID || report.hasOutstandingIOU) {
+ * @param {String} currentUserLogin
+ * @param {Object[]} iouReports
+ * @param {String} iouReports[].ownerEmail
+ * @returns {boolean}
+ */
+function hasOutstandingIOU(report, currentUserLogin, iouReports) {
+    if (!report || !report.iouReportID || report.hasOutstandingIOU) {
+        return false;
+    }
+
+    const iouReport = iouReports && iouReports[`${ONYXKEYS.COLLECTION.REPORT_IOUS}${report.iouReportID}`];
+    if (!iouReport || !iouReport.ownerEmail) {
+        return false;
+    }
+
+    return iouReport.ownerEmail !== currentUserLogin;
+}
+
+/**
+ * Takes several pieces of data from Onyx and evaluates if a report should be shown in the option list (either when searching
+ * for reports or the reports shown in the LHN).
+ *
+ * This logic is very specific and the order of the logic is very important. It should fail quickly in most cases and also
+ * filter out the majority of reports before filtering out very specific minority of reports.
+ *
+ * @param {Object} report
+ * @param {String} currentlyViewedReportID
+ * @param {Boolean} isInGSDMode
+ * @param {String} currentUserLogin
+ * @param {Object} iouReports
+ * @param {String[]} betas
+ * @param {Object} policies
+ * @returns {boolean}
+ */
+function shouldReportBeInOptionList(report, currentlyViewedReportID, isInGSDMode, currentUserLogin, iouReports, betas, policies) {
```

Comment:
NAB.
I'm not sure I love the betas as a parameter. Could we get them from Onyx?